pkg/dockerfile/buildargs.go

```diff
@@ -53,3 +54,15 @@ func (b *BuildArgs) ReplacementEnvs(envs []string) []string {
 	filtered := b.FilterAllowed(envs)
 	return append(envs, filtered...)
 }
+
+func (b *BuildArgs) AddMetaArgs(metaArgs map[string]string) *BuildArgs {
+	for key, value := range metaArgs {
+		logrus.Infof("META ARG: adding %s = %s", key, value)
+		b.AddMetaArg(key, &value)
```

Comment:
I think the bug is here -- since there is a single `value` variable, the address it's pointing to is the same each time, so when that's passed in to AddMetaArg, all of the meta args end up pointing to the same value.

I think if you change it to 
```golang
for key, value := range metaArgs {
  v := value
  b.AddMetaArg(key, &v)
}

``` 
this should work?

Reply:
That was it. Thank you!